packages/vite/src/plugin.ts

```diff
@@ -0,0 +1,57 @@
+import { resolver } from './resolver.js';
+import { hbs } from './hbs.js';
+import { scripts } from './scripts.js';
+import { templateTag } from './template-tag.js';
+import { optimizeDeps } from './optimize-deps.js';
+import { compatPrebuild } from './build.js';
+import { assets } from './assets.js';
+import { contentFor } from './content-for.js';
+import { babel } from '@rollup/plugin-babel';
+import { type ConfigEnv, type UserConfig } from 'vite';
+
+export const extensions = ['.mjs', '.gjs', '.js', '.mts', '.gts', '.ts', '.hbs', '.json'];
+
+export default function ember() {
+  return [
+    {
+      name: 'vite-plugin-ember',
+      enforce: 'pre',
+      config(_config: UserConfig, env: ConfigEnv) {
+        return {
+          resolve: {
+            extensions,
+          },
+          optimizeDeps: optimizeDeps(),
+          server: {
+            port: 4200,
+          },
+          build: {
```

Comment:
I think build, and the choice to configure `tests` should be left to the user.

right now there some unease around `/tests/` and the escape hatch we have in the existing vite config is to encourage changing `/tests/` to `/tests__whatever_i_want/`

Reply:
for this one, specifically, I think it's _better_ to expose always, for discoverability

Reply:
what if you could just do and we have this in the userland vite config? 

```js
ember({
  testUrl: '/tests/`
})
```

Reply:
why invent and maintain a new API tho?

Reply:
(i'll defer to others on this if they disagree with me, ofc)

Reply:
Actually maybe I need to understand this unease and escape hatch.

How would the user configure this, they would also need to rename the tests folder wouldn't they?